diff --git a/dist/app/src/session.js b/dist/app/src/session.js
index 028c7068d08850d72911d42b80396045d95e5650..ba4f5d8fab9162745b2f0fe18f2134d2573028f6 100644
--- a/dist/app/src/session.js
+++ b/dist/app/src/session.js
@@ -21,6 +21,10 @@ export const sessions = withGetter(writable({}));
 export const session = withGetter(derived([pubkey, sessions], ([$pubkey, $sessions]) => ($pubkey ? $sessions[$pubkey] : undefined)));
 export const getSession = (pubkey) => sessions.get()[pubkey];
 export const addSession = (session) => {
+    const previousSession = getSession(session.pubkey);
+    if (previousSession && !equals(previousSession, session)) {
+        getSigner.pop(previousSession)?.cleanup?.();
+    }
     sessions.update(assoc(session.pubkey, session));
     pubkey.set(session.pubkey);
 };
@@ -103,7 +107,9 @@ export const wrapSigner = (signer) => new WrappedSigner(signer, async (method, t
 });
 export const getSigner = cached({
     maxSize: 100,
-    getKey: ([session]) => `${session?.method}:${session?.pubkey}`,
+    getKey: ([session]) => isNip46Session(session)
+        ? `${session.method}:${session.pubkey}:${session.secret}:${session.handler.pubkey}:${session.handler.relays.join(",")}`
+        : `${session?.method}:${session?.pubkey}`,
     getValue: ([session]) => {
         if (isNip07Session(session))
             return wrapSigner(new Nip07Signer());
@@ -117,6 +123,7 @@ export const getSigner = cached({
             const { secret: clientSecret, handler: { relays, pubkey: signerPubkey }, } = session;
             const broker = new Nip46Broker({ clientSecret, signerPubkey, relays });
             const signer = new Nip46Signer(broker);
+            signer.pubkey = session.pubkey;
             return wrapSigner(signer);
         }
     },
